mlx_graphs/utils/utils.py

```diff
@@ -0,0 +1,14 @@
+from typing import Tuple
+
+import mlx.core as mx
+
+
+def gather_src_dst(x: mx.array, edge_index: mx.array) -> Tuple[mx.array, mx.array]:
+    src_idx, dst_idx = edge_index
+    x_i = x[src_idx]
+    x_j = x[dst_idx]
+
+    return x_i, x_j
```

Comment:
I suggest using more readable function and variable names. Right now, this could be interpreted as just gathering indices, while that's not what it's doing.

```suggestion
def map_node_features_to_index(x: mx.array, edge_index: mx.array) -> Tuple[mx.array, mx.array]:
    """
    Extracts source and destination node features based on the given edge indices.

    Parameters:
    - x (mx.array): The input array of node features.
    - edge_index (mx.array): An array of size [2, number_of_edges], where each columns contains the source
                and destination nodes of an edge.

    Returns:
    Tuple[mx.array, mx.array]: A tuple containing source and destination features.
    """
    source_indices, destination_indices = edge_index
    source_features = features[source_indices]
    destination_features = features[destination_indices]

    return source_features, destination_features
```

Reply:
True! I'll modify that